sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -169,7 +170,7 @@ public PBegin begin() {
    * @see #apply(String, PTransform)
    */
   public <OutputT extends POutput> OutputT apply(
-      PTransform<? super PBegin, OutputT> root) {
+      PTransform<? super PBegin, OutputT> root) throws ValidationException {
```

Comment:
Mind being a bit more specific on how the ValidationException should be handled ?